File: packages/StewardApi/StewardApi/Contracts/Pipelines/SimplifiedObligationPipeline.cs
﻿using System.Collections.Generic;

namespace Turn10.LiveOps.StewardApi.Contracts.Pipelines
{
    /// <summary>
    ///     Represents the information required to create an Obligation pipeline.
    /// </summary>
    public sealed class SimplifiedObligationPipeline
    {
        /// <summary>
        ///     Gets or sets the pipeline name.
        /// </summary>
        public string PipelineName { get; set; }

        /// <summary>
        ///     Gets or sets pipeline description.
        /// </summary>
        public string PipelineDescription { get; set; }

        /// <summary>
        ///     Gets or sets the obligation pipelines.
        /// </summary>
        public IList<ObligationKustoDataActivity> KustoDataActivities { get; set; }

        /// <summary>
        ///     Gets or sets the obligation restate-o-matic pipelines.
        /// </summary>
        public IList<ObligationKustoRestateOMaticDataActivity> KustoRestateOMaticDataActivities { get; set; }

        /// <summary>
        ///     Gets or sets principals.
        /// </summary>
        public IList<ObligationPrincipal> Principals { get; set; }
    }
}

File: packages/StewardClient/src/app/shared/models/pipelines/simplified-obligation-pipeline.ts
import { ObligationKustoDataActivity } from './obligation-kusto-data-activity';
import { ObligationKustoRestateOMaticDataActivity } from './obligation-kusto-restate-o-matic-data-activity';
import { ObligationPrincipal } from './obligation-principal';

/** Represents the information required to create an Obligation pipeline. */
export interface SimplifiedObligationPipeline {
  pipelineName: string;
  pipelineDescription: string;
  kustoDataActivities: ObligationKustoDataActivity[];
  kustoRestateOMaticDataActivities: ObligationKustoRestateOMaticDataActivity[];
  principals: ObligationPrincipal[];
}
